src/persistence/serialize.cpp

```diff
@@ -227,52 +95,29 @@ size_t dataToVUint(const QByteArray& data)
 unsigned getVUint32Size(QByteArray data)
 {
     unsigned lensize=0;
-    {
-        unsigned char num3;
-        do {
-            num3 = data[lensize];
-            lensize++;
-        } while ((num3 & 0x80) != 0);
-    }
-    return lensize;
-}
 
-QByteArray uint8ToData(uint8_t num)
-{
-    QByteArray data(1,0);
-    data[0] = (uint8_t)num;
-    return data;
-}
-
-QByteArray uint16ToData(uint16_t num)
-{
-    QByteArray data(2,0);
-    data[0] = (uint8_t)(num & 0xFF);
-    data[1] = (uint8_t)((num>>8) & 0xFF);
-    return data;
-}
+    char num3;
+    do {
+        num3 = data[lensize];
+        lensize++;
+    } while ((num3 & 0x80) != 0);
 
-QByteArray uint32ToData(uint32_t num)
-{
-    QByteArray data(4,0);
-    data[0] = (uint8_t)(num & 0xFF);
-    data[1] = (uint8_t)((num>>8) & 0xFF);
-    data[2] = (uint8_t)((num>>16) & 0xFF);
-    data[3] = (uint8_t)((num>>24) & 0xFF);
-    return data;
+    return lensize;
 }
 
-QByteArray uint64ToData(uint64_t num)
+QByteArray vintToData(int num)
 {
-    QByteArray data(8,0);
-    data[0] = (uint8_t)(num & 0xFF);
-    data[1] = (uint8_t)((num>>8) & 0xFF);
-    data[2] = (uint8_t)((num>>16) & 0xFF);
-    data[3] = (uint8_t)((num>>24) & 0xFF);
-    data[4] = (uint8_t)((num>>32) & 0xFF);
-    data[5] = (uint8_t)((num>>40) & 0xFF);
-    data[6] = (uint8_t)((num>>48) & 0xFF);
-    data[7] = (uint8_t)((num>>56) & 0xFF);
+    QByteArray data(sizeof(int), 0);
+    // Write the size in a Uint of variable lenght (8-32 bits)
+    int i=0;
+    while (num >= 0x80)
+    {
+        data[i] = static_cast<char>(num | 0x80);
+        i++;
```

Comment:
@BiTOk Calm down, this is nitpicking and completely irrelevant here (and anywhere else!!). Fact is: Behaviour is neither standardized nor consistent throughout compilers or even their versions respectively. The ONLY difference between prefix and postfix is simply, that prefix has an immediate effect on the variable, while postfix alters the variable at "end-of-command". I could also write: Use a compiler, that does things right and leave MSC behind. :)